File: app/src/main/java/com/example/spectrumanalyzer/Points.java
package com.example.spectrumanalyzer;

/**
 * Created by usr1 on 2017/05/28.
 */

public class Points {
    private int mMaxNum;
    private int mNum;
    private float[] mX;
    private float[] mY;

    Points(int maxNum) {
        mMaxNum = maxNum;
        mNum = 0;
        mX = new float[maxNum];
        mY = new float[maxNum];
    }

    public void init() {
        mNum = 0;
    }

    public void add(float x, float y) {
        if (mNum < mMaxNum) {
            mX[mNum] = x;
            mY[mNum] = y;
            mNum++;
        }
    }

    public float getX(int index) {
        float ret;
        if ((index >= 0) && (index < mNum)) {
            ret = mX[index];
        } else {
            ret = 0;
        }
        return ret;
    }

    public float getY(int index) {
        float ret;
        if ((index >= 0) && (index < mNum)) {
            ret = mY[index];
        } else {
            ret = 0;
        }
        return ret;
    }

    public int getNum() {
        return mNum;
    }
}

File: app/src/main/java/com/example/spectrumanalyzer/Points.kt
package com.example.spectrumanalyzer

class Points internal constructor(private val mMaxNum: Int) {
    private var mNum: Int = 0
    private val mX: FloatArray
    private val mY: FloatArray

    init {
        mX = FloatArray(mMaxNum)
        mY = FloatArray(mMaxNum)
    }

    fun init() {
        mNum = 0
    }

    fun add(x: Float, y: Float) {
        if (mNum < mMaxNum) {
            mX[mNum] = x
            mY[mNum] = y
            mNum++
        }
    }

    fun getX(index: Int): Float {
        val ret: Float
        ret = if (index >= 0 && index < mNum) {
            mX[index]
        } else {
            0f
        }
        return ret
    }

    fun getY(index: Int): Float {
        val ret: Float
        ret = if (index >= 0 && index < mNum) {
            mY[index]
        } else {
            0f
        }
        return ret
    }

    fun getNum(): Int {
        return mNum
    }
}
